### PR TITLE
Add npx project setup option

### PR DESCRIPTION
**Pull request by edakturk14**
Adding the npx setup option. 

1. Since this is beta its also not on the original se2 readme, should I make an update there as well, or do we want to hold off to update. 
2. I like that there is a quickstart thats one section however I was thinking that we could split the QuickStart into two sections: Installation (Clone and npx) and Getting Started (yarn chain etc.). Would love some ideas on this, I can make the changes if it make sense to others as well.

### CODE REVIEW

**Comment by technophile-04** [comment]
Tysm Eda  🙌 !!

> Since this is beta its also not on the original se2 readme, should I make an update there as well, or do we want to hold off to update.

Just created https://github.com/scaffold-eth/scaffold-eth-2/issues/508, maybe we can have a CLI section in the bottom of SE-2(`main`) readme with a 2-3 line description and there we can link people to `npx` section of docs I (once this PR is merged) ? 

> I like that there is a quickstart thats one section however I was thinking that we could split the QuickStart into two sections: Installation (Clone and npx) and Getting Started (yarn chain etc.). Would love some ideas on this, I can make the changes if it make sense to others as well.

Yup makes sense!!. Alternatively also I was wondering what if we have it like this in the left sidebar? instead on single page : 

Quickstart  
  1. Setup using clone
  2. using `npx`

because having this will allow us maybe add a bit more details for `foundry` folder structure like we have it in [CLI README QuickStart](https://github.com/scaffold-eth/scaffold-eth-2/tree/cli#quickstart)

just thinking out loud, but would love to know other's thought too 🙌 

cc @Pabl0cks

File: docs/quick-start.md [hunk 1]
```diff
@@ -52,4 +57,43 @@ Run smart contract test with `yarn hardhat:test`
...
+1. Bootstrap the project:
+```
+npx create-eth@latest
+```
+
+You will be presented with a series of prompts:
+
+- Your project name: Enter a name for your project, e.g., my-dapp-example.
+- What solidity framework do you want to use?: Choose your preferred solidity framework (Hardhat, Foundry)
+- Install packages?: Confirm if you want to install the necessary packages, typically by selecting Yes.
```
Comment (technophile-04) [inline docs/quick-start.md:75]: Screenshot 2023-08-28 at 1 12 31 PM

`nitpick:` I have seen begginers getting confused about how to select this, maybe we could say `hit enter` to select `yes` (defualt option) or type `n` and hit enter to select to `no` 

but lol yeah we could skip this if its too obvious

**Comment by Pabl0cks** [comment]
Nice job @edakturk14!! 🙌

I see a great idea to separate Installation in a different subsection now that we got 2 options for it, so is a bit more isolated and we don't have to replicate the common parts on both options (yarn chain, yarn deploy, yarn start). 

Maybe instead of "Getting Started" we can call that second subsection "Environment" like we did on the old SRE (we had Install+Environment sections), even if in new SRE-SE2 we merged both in a single "Environment" section, now I feel in here it makes sense to have Installation + Environment separated.

What do you think in some structure like this?:

# QuickStart
## Installation
### Requirements
### Option 1: Setup using git clone
### Option 2: [Beta] Setup using npx create-eth@latest
## Environment
Now that our installation is complete, we will set up our development environment for Scaffold ETH-2, which consists of three terminals:

1. Run a local network in the first terminal....

File: docs/quick-start.md (hunk 1 above)
Comment (edakturk14) [reply docs/quick-start.md:75]: great suggestion!! Will be changing this!

**Comment by edakturk14** [comment]
Thank you both @technophile-04 & @Pabl0cks!! These are great suggestions! Let me work on the changes today. 

also definitely  agree on updating the main se-2 readme once this PR is merged. We could have the same content there, instead of side bars it would be subheaders in that case - and we can make any iterations as needed.

**Review by Pabl0cks — COMMENTED** [review]
I see changes nice! 🙌 Added some small comments. 

Question for all (specially for future sections): Having now QuickStart parent not clickable I had doubts if users may get confused, but watching Docusaurus docs I see they also have some parents not clickable so I guess we will be fine :) Would love to hear your opinion too!

Question to @technophile-04: Do you think we should also work on adding Foundry explanations from cli branch into Docs right now? Or should we wait a bit?


Thanks ♥

**Comment by technophile-04** [comment]
Tysm Eda !! Looking great ! 

 > Added a "What's next" section on the end of the environment (https://scaffold-eth-2-docs-ciuxg2nes-buidlguidldao.vercel.app/quick-start/environment#whats-next), we can change the name (eg: Further steps?)
 
I love the idea of this section !! 
 
Maybe we could rearrange it a bit : 

- Edit your smart contract `YourContract.sol` in `packages/hardhat/contracts`
- Edit your deployment scripts in `packages/hardhat/deploy`
- Edit your frontend in `packages/nextjs/pages`
- Edit the app config in `packages/nextjs/scaffold.config.ts`
- Run smart contract test with `yarn hardhat:test`

> Question for all (specially for future sections): Having now QuickStart parent not clickable I had doubts if users may get confused, but watching Docusaurus docs I see they also have some parents not clickable so I guess we will be fine :) Would love to hear your opinion too!

Yeah make sense! Just to keep consistency overall with others maybe we can do something similar to https://docs.scaffoldeth.io/category/-shipping-your-dapp ? What do you all think ? 


> Do you think we should also work on adding Foundry explanations from cli branch into Docs right now? Or should we wait a bit?


Umm yeah I was actually thinking about it how we can add it in exsisting docs, maybe i will create an issue regarding adding Foundry explanantion in this docs once this https://github.com/scaffold-eth/scaffold-eth-2/pull/489 PR is done  🙌

**Comment by edakturk14** [comment]
Great suggestions - I've made the grammar/typo + rearranging changes! Thanks a lot!!

> Yeah make sense! Just to keep consistency overall with others maybe we can do something similar to https://docs.scaffoldeth.io/category/-shipping-your-dapp ? What do you all think ?

Yeah now looking into it I agree that it can be confusing that the Quickstart it not clickable when the others are. I like the suggestion very much! What do you think @Pabl0cks?

> Umm yeah I was actually thinking about it how we can add it in exsisting docs, maybe i will create an issue regarding adding Foundry explanantion in this docs once this https://github.com/scaffold-eth/scaffold-eth-2/pull/489 PR is done 🙌

sounds great!

**Comment by Pabl0cks** [comment]
> Yeah now looking into it I agree that it can be confusing that the Quickstart it not clickable when the others are. I like the suggestion very much! What do you think @Pabl0cks?

Yeah! That was my natural path, didn't feel rush since others were mixing clickable/non clickable, but for now I'd also continue having all sections clickable until we find a reason not to 😊🙌

**Comment by edakturk14** [comment]
made the updates, lmk what you think and happy to look into the missing parts @technophile-04 & @Pabl0cks

**Comment by Pabl0cks** [comment]
> made the updates, lmk what you think and happy to look into the missing parts @technophile-04 & @Pabl0cks

Everything is cool for me! GJ Eda 🙌

I've just pushed a small tweak to make QuickStart clickable. 

Just learnt about `description` parameter to set a custom card description, instead of autogenerated from page content. Feel free to adapt those descriptions I've added for others you prefer 😊

**Review by Pabl0cks — APPROVED** [review]

**Comment by technophile-04** [comment]
Tysm Eda !!! and Thanks Pablo for the review !!, its looking really nice !! Merging this 🙌

**Pull request merged**